### PR TITLE
Fix bug in root repo import where catalog-info.yaml.hcl file breaks import

### PR DESCRIPTION
**Pull request by sblausten**
## Hey, I just made a Pull Request!
When importing a repository from GitHub using the root repo url, if the repo has any templated catalog-info files (as is the case with the backstage repo itself, the search analyser finds the .hcl file and attempts to import it as a location. This breaks the import with a unclear 500 error
336828088-9d8e44f2-0bec-4415-a05d-6f57e1571296

Old pr https://github.com/backstage/backstage/pull/25071

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by sblausten** [comment]
@Rugvip Sorry I started fresh because of the signed off thing. This takes into account your comment on file extensions potentially not being yaml/yml

File: plugins/catalog-backend-module-github/src/analyzers/GithubLocationAnalyzer.ts [hunk 1]
```diff
@@ -76,8 +76,13 @@ export class GithubLocationAnalyzer implements ScmLocationAnalyzer {
     const { owner, name: repo } = parseGitUrl(url);
 
     const catalogFile = catalogFilename || 'catalog-info.yaml';
+    const fileParts = catalogFile.split('.');
+    const extension =
+      fileParts.length > 0
```
[the remark this answers is not in the document]
Comment (sblausten) [reply plugins/catalog-backend-module-github/src/analyzers/GithubLocationAnalyzer.ts:81]: Not sure I follow - if there is no extension in the file name passed in, we don't add an extension filter at all i.e. `extension:yaml;`. `fileParts.length > 1` seems to me to exclude most file names so we will just not be passing the extension filter in most scenarios which wont fix the bug - i.e. in backstage repo's case, the file is `catalog-info.yaml` and `fileParts.length > 1` will be false so the extension variable will end up as an empty string. This means the query will be the same is it currently is and the hcl file will still appear in results and break the import.

[the remark this answers is not in the document]
Comment (freben) [reply plugins/catalog-backend-module-github/src/analyzers/GithubLocationAnalyzer.ts:81]: Maybe I'm just being dense. But

```js
'catalog-info.yaml'.split('.').length
```

equals 2 :) Which is what I meant by `> 1`.

Another probably wiser update could be to `import { extname } from 'path'` and use that to extract the extension.  Gotta remove leading dot then of course. That one is smart enough to know that for example the file `.gitignore` has no extension at all.

[the remark this answers is not in the document]
Comment (sblausten) [reply plugins/catalog-backend-module-github/src/analyzers/GithubLocationAnalyzer.ts:81]: Ah sorry yes I see what you mean now

[the remark this answers is not in the document]
Comment (sblausten) [reply plugins/catalog-backend-module-github/src/analyzers/GithubLocationAnalyzer.ts]: I made that change

**Comment by sblausten** [comment]
I believe the failing test at `plugins/proxy-backend/src/service/router.credentials.test.ts` is unrelated - could someone try a re-run?

**Review by freben — APPROVED** [review]

**Pull request merged**